### PR TITLE
Ignore TLS/SSL error for Redis

### LINKED ISSUE #3871: [redis] Ignore TLS/SSL error
### ⚠️ Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find similar feature request

### 🏷️ Feature Request Type

UI Feature, Other

### 🔖 Feature description

Provide the capability to ignore the TLS/SSL error that is returned by a Redis service.  
`self-signed certificate in certificate chain`

Heroku recommends this as a way to connect in Node.js https://devcenter.heroku.com/articles/connecting-heroku-redis#connecting-in-node-js.

### ✔️ Solution

The `redis` package allows us to set `rejectUnauthorized: false` through the `socket` [createClient configuration property](https://github.com/redis/node-redis/blob/master/docs/client-configuration.md#create-a-ssl-client-using-a-self-signed-certificate).

We can utilise the existing monitor `ignoreTls` variable to include the same input that is part of the Additional section when the `monitor.type` is `http | keyword | json-query`. It can then be utilised by the `redisPingAsync` method.

Consideration will have to be taken as the current `ignoreTLSError` translations refer specifically to 'HTTPS websites'

### ❓ Alternatives

_No response_

### 📝 Additional Context

_No response_

### PR DESCRIPTION
**Pull request by HdroguettA**
⚠️⚠️⚠️ Since we do not accept all types of pull requests and do not want to waste your time. Please be sure that you have read pull request rules:
https://github.com/louislam/uptime-kuma/blob/master/CONTRIBUTING.md#can-i-create-a-pull-request-for-uptime-kuma

Tick the checkbox if you understand [x]:
- [x] I have read and understand the pull request rules.

# Description

Fixes #3871 

## Type of change

- User interface (UI)
- Other

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I ran ESLint and other linters for modified files
- [x] I have performed a self-review of my own code and tested it
- [x] I have commented my code, particularly in hard-to-understand areas
  (including JSDoc for methods)
- [x] My changes generate no new warnings
- [ ] My code needed automated testing. I have added them (this is optional task)

### CODE REVIEW

**Comment by louislam** [comment]
It did not pass the linter.

File: src/pages/EditMonitor.vue [hunk 1]
```diff
@@ -481,10 +481,10 @@
                                 </div>
                             </div>
 
-                            <div v-if="monitor.type === 'http' || monitor.type === 'keyword' || monitor.type === 'json-query' " class="my-3 form-check">
+                            <div v-if="monitor.type === 'http' || monitor.type === 'keyword' || monitor.type === 'json-query' || monitor.type === 'redis' " class="my-3 form-check">
                                 <input id="ignore-tls" v-model="monitor.ignoreTls" class="form-check-input" type="checkbox" value="">
                                 <label class="form-check-label" for="ignore-tls">
-                                    {{ $t("ignoreTLSError") }}
+                                    {{ ignoreTlsLabel }}
```
Comment (CommanderStorm) [inline src/pages/EditMonitor.vue:487]:
```suggestion
                                    {{ this.monitor.type === "redis" ? this.$t("ignoreTLSErrorGeneral") : this.$t("ignoreTLSError") }}
```

File: src/pages/EditMonitor.vue [hunk 2]
```diff
@@ -1201,6 +1201,16 @@ message HealthCheckResponse {
...
+        /**
+         * Retrieves the appropriate ignore TLS error label based on the monitor type.
+         * @returns {string} Returns the non specific label if the monitor type is 'redis', otherwise, returns the default.
+         */
+        ignoreTlsLabel() {
+            return this.monitor.type === "redis"
+                ? this.$t("ignoreTLSErrorGeneral")
+                : this.$t("ignoreTLSError");
+        }
+
```
Comment (CommanderStorm) [inline src/pages/EditMonitor.vue:1213]:
```suggestion
```

File: src/pages/EditMonitor.vue [hunk 3]
```diff
@@ -1217,6 +1227,7 @@ message HealthCheckResponse {
             }
         }
     },
+
```
Comment (CommanderStorm) [inline src/pages/EditMonitor.vue:1230]:
```suggestion
```

**Review by CommanderStorm — COMMENTED** [review]
this is an attempt to fix the linting error

File: src/pages/EditMonitor.vue [hunk 4]
```diff
@@ -482,10 +482,10 @@
                                 </div>
                             </div>
 
-                            <div v-if="monitor.type === 'http' || monitor.type === 'keyword' || monitor.type === 'json-query' " class="my-3 form-check">
+                            <div v-if="monitor.type === 'http' || monitor.type === 'keyword' || monitor.type === 'json-query' || monitor.type === 'redis' " class="my-3 form-check">
                                 <input id="ignore-tls" v-model="monitor.ignoreTls" class="form-check-input" type="checkbox" value="">
                                 <label class="form-check-label" for="ignore-tls">
-                                    {{ $t("ignoreTLSError") }}
+                                    {{ this.monitor.type === "redis" ? this.$t("ignoreTLSErrorGeneral") : this.$t("ignoreTLSError") }}
```
Comment (CommanderStorm) [inline src/pages/EditMonitor.vue:488]: I apparently missed this linting issue

```suggestion
                                    {{ monitor.type === "redis" ? $t("ignoreTLSErrorGeneral") : $t("ignoreTLSError") }}
```

**Review by CommanderStorm — APPROVED** [review]
Thanks for the improvement!

All changes in this PR are small and uncontroversial
⇒ merging with junior maintainer approval

**Pull request merged**